src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -467,49 +394,43 @@ export function fillLoadoutFromEquipped(
     const newEquippedItems = store.items.filter(
       (item) => item.equipped && itemCanBeInLoadout(item) && itemMatchesCategory(item, category)
     );
-    const mods: number[] = [];
+    const modsByBucket: { [bucketHash: number]: number[] } = {};
     for (const item of newEquippedItems) {
       if (!(item.bucket.hash in equippedItemsByBucket)) {
-        const loadoutItem = convertToLoadoutItem(item, true, 1);
-        if (item.bucket.hash === BucketHashes.Subclass) {
-          loadoutItem.socketOverrides = createSocketOverridesFromEquipped(item);
+        loadout = addItem(defs, item, true)(loadout);
+
+        // Only save fashion for the items we added
+        const plugs = item.sockets
+          ? filterMap(
+              getSocketsByCategoryHash(item.sockets, SocketCategoryHashes.ArmorCosmetics),
+              (s) => s.plugged?.plugDef.hash
+            )
+          : [];
+        if (plugs.length) {
+          modsByBucket[item.bucket.hash] = plugs;
         }
-        loadout.items.push(loadoutItem);
-        mods.push(...extractArmorModHashes(item).map(mapToNonReducedModCostVariant));
       }
     }
-    if (mods.length && (loadout.parameters?.mods ?? []).length === 0) {
-      loadout.parameters = {
-        ...loadout.parameters,
-        mods,
-      };
-    }
-    if (artifactUnlocks?.unlockedItemHashes.length) {
-      loadout.parameters = {
-        ...loadout.parameters,
-        artifactUnlocks,
-      };
+
+    // Populate mods if they aren't already there
+    if (!category && _.isEmpty(loadout.parameters?.mods)) {
+      loadout = syncModsFromEquipped(store)(loadout);
     }
```

Comment:
This is a behavior change - previously an empty mods list would be replaced with the mods from  associated with any newly-added armor item. Now, it syncs all the mods from equipped items if the mods list is empty, and won't do anything if you're just clicking "fill in using equipped" on the Armor section.